raw_data/preprocessed/africa/rwanda_french.txt

```diff
@@ -0,0 +1,303 @@
+CHAPITRE PREMIER - DES DISPOSITIONS GENERALES 
+
+(1) La présente loi organique détermine les modalités de protéger, sauvegarder et promouvoir l'environnement au Rwanda. La présente loi organique vise à: [1] protéger l'environnement, la population et son habitat; [2] établir les principes fondamentaux relatifs à la protection de l'environnement contre toutes les formes de dégradation afin de valoriser les ressources naturelles, de lutter contre toutes sortes de pollutions et de nuisances; [3] améliorer les conditions de vie de la population dans le respect de l'équilibre avec les ressources disponibles; [4] assurer la durabilité des ressources dans le respect du même droit entre les générations présentes et futures; [5] garantir à tous les Rwandais un développement économique durable, qui ne porte pas atteinte à l'environnement et au bien-être de la population; [6] mettre en place les mécanismes de protection et d'atténuation des effets négatifs sur l'environnement et veiller à la restauration des milieux endommagés.
+
+(2) L'environnement au Rwanda constitue un patrimoine commun de la Nation et une partie intégrante du patrimoine mondial.
+
+(3) Toute personne a le devoir de protéger, sauvegarder et promouvoir l'environnement. L'État est le garant de la protection, la sauvegarde et la promotion de l'environnement.
+
+CHAPITRE II - DES DEFINITIONS DE QUELQUES MOTS UTILISES DANS LA PRESENTE LOI ORGANIQUE
+
+(4) Aux termes de la présente loi organique: L'environnement est l'ensemble des éléments composés de l'environnement naturel et de l'environnement humain. On y trouve les éléments chimiques, la diversité biologique ainsi que des facteurs socio-économiques, culturels, esthétiques, scientifiques susceptibles d'avoir un effet direct ou indirect, immédiat ou à terme, sur le développement du milieu, des êtres vivants et des activités humaines. 
```

Comment:
I'm not sure how to remove all bracketed numbers within just article 4. I've tried `sed -E ':start;s/^(\(4\).*) [0-9]+ /\1/;t start' | \`

But it's not working, any suggestions?

Reply:
I'm having some difficulty removing the bracketed numbers from article 4. I've tried `sed -E ':start;s/^(\(4\).*) \[[0-9]+\] /\1/;t start'` but it's not working, any suggestions?

So if the bracketed letters are preceded by a period, does this mean they are not forming a continuous sentence and should be deleted?

Reply:
The problem is likely that when you invoke that sed command (which is otherwise properly constructed), article 4 has already been split over a number of lines. Remember that sed processes the input line by line, so the `^(\(4\).*)` part of your regular expression will only match line 11 and nothing thereafter. I should mention that I want everything corresponding to article 4 to end up on one line. Confining everything to a single line is necessary for subsequent article-to-article lexical comparisons (the job of the Python code in the 'eia' project).

If a bracketed letter is preceded by a period, then yes, the text that follows is not part of a longer sentence and the bracketed letter should be deleted.

Reply:
Thanks for the clarification. The issue ended up being about the ordering of the formatting.